src/containers/Projects/index.js

```diff
@@ -1,54 +1,87 @@
-// TODO: add flow annotations
+/* @flow */
 
 import React, { Component } from 'react'
 import Helmet from 'react-helmet'
 import { push } from 'react-router-redux'
 import CircularProgress from 'material-ui/CircularProgress'
-import { ErrorMessage, ProjectList } from 'components'
 import { connect } from 'react-redux'
-import { fetchProjects } from 'ducks/projects'
 import { Sticky, StickyContainer } from 'react-sticky'
+import urljoin from 'url-join'
+
+import ProjectList from 'components/ProjectList/ProjectList'
+import ErrorMessage from 'components/ErrorMessage'
+import GroupList from 'components/GroupList/GroupList'
+import { fetchProjects, clearProjects } from 'ducks/projects'
 import { Row, Col, Button, ButtonGroup } from 'react-bootstrap'
+import { helpers, groupPathFromPath } from 'routes/helpers'
+
 
 export type Props = {
   isFetching: boolean,
-  errors?: Array<any>,
+  errors: Array<any>,
   projects: Array<any>,
   dispatch: Function,
   theme: Object,
+  pathname: string,
+  tree: any,
 };
 
 export class Projects extends Component {
+
   componentDidMount() {
-    const { dispatch } = this.props
-    this.clickHandler = this.clickHandler.bind(this)
-    // todo: remove this from constructor
-    dispatch(fetchProjects())
+    const { dispatch, pathname } = this.props
+    this.projectClickHandler = this.projectClickHandler.bind(this)
+    this.groupClickHandler = this.groupClickHandler.bind(this)
+    this.props.dispatch(clearProjects())
+    this.updateNode(dispatch, pathname)
+  }
+
+  componentWillReceiveProps(nextProp, nextState) {
+    if (this.props.pathname !== nextProp.pathname) {
+      this.props.dispatch(clearProjects())
+      this.updateNode(this.props.dispatch, nextProp.pathname)
+    }
+  }
+
+  updateNode(dispatch, pathname) {
+    dispatch(fetchProjects(groupPathFromPath(pathname)))
   }
 
   props: Props
+  groupClickHandler(value: string) {
+    const { dispatch, pathname } = this.props
+    const base = urljoin(pathname.split('/').slice(0, -1))
+    let path
+    if (value === 'Parent Group') {
```

Comment:
This constant `Parent Group` would be nice to define once. There is a relationship between this and the one that you add inside the graphql query.